src/ampycloud/utils/utils.py

```diff
@@ -200,3 +202,38 @@ def tmp_seed(seed : int):
         yield
     finally:
         np.random.set_state(state)
+
+
+@log_func_call(logger)
+def adjust_nested_dict(ref_dict: dict, new_dict: dict, lvls: list = None) -> dict:
+    """ Update a given (nested) dictionnary given a second (possibly incomplete) one.
+
+    Args:
+        ref_dict (dict): reference dict of dict (of dict of dict ...).
+        new_dict (dict): values to update as a dict (of dict or dict of dict ...)
+        lvls (list, optional): levels of the previous nested dict layers, used for reporting useful
```

Comment:
what are the possible values of lvls? it is not very clear to me how to use it.

Reply:
This is used by the function itself when it calls itself. There is no need for the user to set this to anything at first.